reactivesocket-core/src/main/java/io/reactivesocket/server/ReactiveSocketServer.java

```diff
@@ -42,24 +45,28 @@
 
     static ReactiveSocketServer create(TransportServer transportServer) {
         return acceptor -> {
-            return transportServer.start(duplexConnection -> {
-                return Px.from(duplexConnection.receive())
+            return transportServer.start(connection -> {
+                return Px.from(connection.receive())
                   .switchTo(setupFrame -> {
                       if (setupFrame.getType() == FrameType.SETUP) {
+                          ClientServerInputMultiplexer multiplexer = new ClientServerInputMultiplexer(connection);
                           ConnectionSetupPayload setupPayload = ConnectionSetupPayload.create(setupFrame);
-                          ClientReactiveSocket sender = new ClientReactiveSocket(duplexConnection,
+                          ClientReactiveSocket sender = new ClientReactiveSocket(multiplexer.asServerConnection(),
                                                                                  Throwable::printStackTrace,
                                                                                  StreamIdSupplier.serverSupplier(),
                                                                                  KeepAliveProvider.never());
+                          LeaseHonoringSocket lhs = new DefaultLeaseHonoringSocket(sender);
```

Comment:
Does it even make sense for a client responder to use leases? Either it can take requests or it can't (reject/fail). For example, a server-mobile relationship doesn't make sense for leases, as once a mobile client is connected, it only has that one connection to send requests over. Leases really only make sense in server-server situations where a requesting server has multiple connections open and can send requests to different servers depending on their leases.